tracee/tracee.go

```diff
@@ -254,13 +213,60 @@ func (t *Tracee) initBPF() error {
 	return nil
 }
 
+// Run starts the trace. it will run until interrupted
+func (t Tracee) Run() error {
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+	t.printer.Preamble()
+	t.bpfPerfMap.Start()
+	go t.processEvents()
+	<-sig
+	t.bpfPerfMap.Stop() //TODO: should this be in Tracee.Close()?
+	t.printer.Epilogue()
+	t.Close()
+	return nil
+}
+
+// Close cleans up created resources
+func (t Tracee) Close() {
+	if t.bpfModule != nil {
+		t.bpfModule.Close()
+	}
+}
+
 func boolToUInt32(b bool) uint32 {
 	if b {
 		return uint32(1)
 	}
 	return uint32(0)
 }
 
+// shouldPrintEvent whether or not the given event id should be printed to the output
+func (t Tracee) shouldPrintEvent(e int32) bool {
+	eName := EventsIDToName[e]
+	// first, alsways print non-essential events
+	// if we got a trace for a non-essential event, it means the user explicitly requested it (using `-e`), or the user doesn't care (trace all by default). In both cases it's ok to print.
+	// for essential events we need to check if the user actually wanted this event
+	isEssential := false
+	for _, essentialEvent := range essentialEvents {
+		if eName == essentialEvent {
+			isEssential = true
+			break
+		}
+	}
+	if !isEssential {
+		return true
+	}
+
```

Comment:
If we reached this point, it means the event is an essential event.
As the number of the essential events is only 3, it is better to save a boolean for each of these ones once during init.
This will save the need to iterate through EventsToTrace which can be a big list

Reply:
I have changed how shouldPrintEvent works altogether in https://github.com/aquasecurity/tracee/pull/47/commits/95dbdbe8586281a6ac4aaea089b4a15303b4f8b5. What do you think?

Reply:
Looks good, but I added a comment there